test/runtests.jl

```diff
@@ -1,80 +1,57 @@
 using PowerModels
-import InfrastructureModels
-import Memento
-
-# Suppress warnings during testing.
-Memento.setlevel!(Memento.getlogger(InfrastructureModels), "error")
-PowerModels.logger_config!("error")
+using Test
 
 import HiGHS
+import InfrastructureModels
 import Ipopt
-import SCS
-import Juniper
-
-import JuMP
 import JSON
-
+import JuMP
+import Juniper
 import LinearAlgebra
+import Memento
+import SCS
 import SparseArrays
-using Test
 
+# Suppress warnings during testing.
+Memento.setlevel!(Memento.getlogger(InfrastructureModels), "error")
+PowerModels.logger_config!("error")
 
 # default setup for solvers
-nlp_solver = JuMP.optimizer_with_attributes(Ipopt.Optimizer, "tol"=>1e-6, "print_level"=>0)
-nlp_ws_solver = JuMP.optimizer_with_attributes(Ipopt.Optimizer, "tol"=>1e-6, "mu_init"=>1e-4, "print_level"=>0)
+nlp_solver = JuMP.optimizer_with_attributes(
+    Ipopt.Optimizer,
+    "tol"=>1e-6,
+    "print_level"=>0,
+)
+
+nlp_ws_solver = JuMP.optimizer_with_attributes(
+    Ipopt.Optimizer,
+    "tol"=>1e-6,
+    "mu_init"=>1e-4,
+    "print_level"=>0,
+)
+
+milp_solver =
+    JuMP.optimizer_with_attributes(HiGHS.Optimizer, "output_flag"=>false)
+
+minlp_solver = JuMP.optimizer_with_attributes(
+    Juniper.Optimizer,
+    "nl_solver"=>JuMP.optimizer_with_attributes(
+        Ipopt.Optimizer,
+        "tol"=>1e-4,
+        "print_level"=>0,
+    ),
+    "log_levels"=>[],
+)
 
-milp_solver = JuMP.optimizer_with_attributes(HiGHS.Optimizer, "output_flag"=>false)
-minlp_solver = JuMP.optimizer_with_attributes(Juniper.Optimizer, "nl_solver"=>JuMP.optimizer_with_attributes(Ipopt.Optimizer, "tol"=>1e-4, "print_level"=>0), "log_levels"=>[])
 sdp_solver = JuMP.optimizer_with_attributes(SCS.Optimizer, "verbose"=>false)
 
 include("common.jl")
 
-@testset "PowerModels" begin
-
-    include("matpower.jl")
-
-    include("pti.jl")
-
-    include("psse.jl")
-
-    include("io.jl")
-
-    include("output.jl")
-
-    include("modify.jl")
-
-    include("data.jl")
-
-    include("data-basic.jl")
-
-    include("model.jl")
-
-    include("am.jl")
-
-    include("opb.jl")
-
-    include("pf.jl")
-
-    include("pf-native.jl")
-
-    include("opf.jl")
-
-    include("opf-var.jl")
-
-    include("opf-obj.jl")
-
-    include("opf-ptdf.jl")
-
-    include("ots.jl")
-
-    include("tnep.jl")
-
-    include("multinetwork.jl")
-
-    include("util.jl")
-
-    include("warmstart.jl")
-
-    include("docs.jl")
-
+@testset verbose=true "PowerModels" begin
+    @testset verbose=true "$file" for file in readdir(@__DIR__)
+        if !endswith(file, ".jl") || file in ("common.jl", "runtests.jl")
+            continue
+        end
+        include(joinpath(@__DIR__, file))
+    end
```

Comment:
I prefer the explicit list of imports for the development workflow (which may be outdated), lets discuss that and if this this last change should be reverted.

Reply:
I've found that explicit list of importing test files is very brittle. It's quite easy to miss including test file, and it can encourage people to run test files independently, which can lead to dependent state between the test files.

I would be in favor of moving to something like the JuMP tests, which strongly encapsulate tests into modules and functions to remove all shared state: https://github.com/jump-dev/JuMP.jl/blob/master/test/test_expr.jl. It's quite a large change though.